### ISSUE TITLE
MC sampling  replace IB(information bottleneck) in the code

### ISSUE THREAD

**Issue by wwlaoxi** [ZiweiWangTHU/BiDet#14]
Hi, Wang. Thanks for your great work. I have some doubts,Can you explain the following doubts?

The IB loss in the paper is replace MC sampling int the code.
1.What is the meaning of MC sampling? Is it Monte Carlo （蒙特卡洛）.
2.What is the relationship between MC sampling and IB principle?
3.The feature maps are L2 normalization in the code。What is the relationship between L2 normalization and MC sampling？
 
Thank you

[MC sampling](https://github.com/ZiweiWangTHU/BiDet/issues/3)

**Comment by Wuziyi616** [comment]
Hi! Thank you for your interest in our work! Yes, MC means Monte Carlo sampling. There may not be a theoretical relationship between MC and IB, here we just use MC to calculate the loss. So, if you look at the loss function (which is the image in the issue you referred), it is derived from the IB objective, but really difficult to implement and compute. On the other hand, MC sampling is an easy way to approximate the probability terms and compute the loss which can be back-propagated and used for model training. You can just regard this as a trick used in the implementation.

As for the L2 normalization of feature maps, I do not think I do so in the code, maybe you can refer the exact lines in the code? The only normalization I conduct is over the scores for computing the sparse-object-prior loss [here](https://github.com/ZiweiWangTHU/BiDet/blob/master/ssd/train_bidet_ssd.py#L277-L283). And for the feature map, we only minimize its L2 norm as the IB loss (you can think of it as a regularizer) [here](https://github.com/ZiweiWangTHU/BiDet/blob/master/ssd/train_bidet_ssd.py#L290-L298).
